src/hooks/useCategories.ts

```diff
@@ -1,15 +1,46 @@
-import { CategoryDto } from 'dtos/CategoryDto';
-import { useQuery, UseQueryResult } from 'react-query';
-import CategoryService from 'services/categoryService';
+import CategoryDto from 'dtos/Category/CategoryDto';
+import CategoryProblemDetails from 'dtos/Category/CategoryProblemDetails';
+import CategoryRequestDto from 'dtos/Category/CategoryRequestDto';
+import { useMutation, UseMutationResult, useQuery, UseQueryResult } from 'react-query';
+import categoryService from 'services/categoryService';
+
+interface UpdateCategoryProps {
+  id: number;
+  requestDto: CategoryRequestDto;
+}
 
 interface UseCategoriesResult {
   categories: UseQueryResult<CategoryDto[]>;
+  createCategory: UseMutationResult<CategoryDto, CategoryProblemDetails, CategoryRequestDto>;
+  editCategory: UseMutationResult<CategoryDto, CategoryProblemDetails, UpdateCategoryProps>;
+  removeCategory: UseMutationResult<void, CategoryProblemDetails, number>;
 }
 
 export default function useCategories(filtered = false): UseCategoriesResult {
-  const categories = useQuery<CategoryDto[]>(['categories'], CategoryService.getAll, {
+  const categories = useQuery<CategoryDto[]>(['categories'], categoryService.getAll, {
     select: (data) => (filtered ? data.filter((category) => !category.retired || (category.products && category.products.length > 0)) : data),
   });
 
-  return { categories };
+  const createCategory = useMutation<CategoryDto, CategoryProblemDetails, CategoryRequestDto>((requestDto) => categoryService.create(requestDto), {
+    onSuccess: () => {
```

Comment:
Is it worth refactoring the onSuccess result and refetch into it's own utility class? We seem to use it everywhere